src/queryResult/queryResultWebViewController.ts

```diff
@@ -87,4 +187,47 @@ export class QueryResultWebviewController extends ReactWebviewViewController<
     ): void {
         this._sqlOutputContentProvider = provider;
     }
+
+    public setExecutionPlanService(service: ExecutionPlanService): void {
+        this.executionPlanService = service;
+    }
+
+    public setUntitledDocumentService(
+        service: UntitledSqlDocumentService,
+    ): void {
+        this.untitledSqlDocumentService = service;
+    }
+
+    private async createExecutionPlanGraphs() {
+        if (!this.state.executionPlan) {
+            const planFile: ep.ExecutionPlanGraphInfo = {
+                graphFileContent: this._executionPlanContents,
+                graphFileType: ".sqlplan",
+            };
+            try {
+                this.state.executionPlan =
+                    await this.executionPlanService.getExecutionPlan(planFile);
+                this.state.executionPlanGraphs =
+                    this.state.executionPlan.graphs;
+                this.state.loadState = ApiStatus.Loaded;
+                this.state.totalCost = this.calculateTotalCost();
+            } catch (e) {
+                this.state.loadState = ApiStatus.Error;
+                this.state.errorMessage = e.toString();
+            }
+        }
+        this.updateState();
+    }
+
+    private calculateTotalCost(): number {
+        let sum = 0;
+        for (const graph of this.state.executionPlanGraphs!) {
```

Comment:
Doing `executionPlanGraphs!` without confirming that it's defined makes me nervous.  What about having an `if (undefined) return 0;` before the summation?